packages/react-server/src/ReactFizzServer.js

```diff
@@ -422,8 +422,10 @@ function fatalError(request: Request, error: mixed): void {
   // This is called outside error handling code such as if the root errors outside
   // a suspense boundary or if the root suspense boundary's fallback errors.
   // It's also called if React itself or its host configs errors.
-  const onErrorShell = request.onErrorShell;
-  onErrorShell(error);
+  const onShellError = request.onShellError;
+  onShellError(error);
+  const onFatalError = request.onFatalError;
+  onFatalError(error);
```

Comment:
I forgot to call this in the previous PR. I couldn't actually come up with a way to test this using only public APIs. Because it is very unlikely to happen.

If the shell errors, we error the promise before we can listen to this. If something errors in the subtree, it's recoverable. If something errors in the stream, it's canceled.

So basically the only things that can even trigger this is bugs in React or bugs in the Stream implementation. Since even the React implementation is pretty self-enclosed you can't even patch anything. The streams themselves also protect against this happening due to user code. I think maybe we could use a broken polyfill to test this.